File: PO_py/program/Organisms/Plants/Guarana.py
from program.Organisms.Plant import Plant


class Guarana(Plant):

    GUARANA_STRENGTH = 0
    GUARANA_SEEDING_PROBABILITY = 10

    def __init__(self, world, x, y):
        super().__init__(world, x, y, Guarana.GUARANA_STRENGTH, '^', Guarana.GUARANA_SEEDING_PROBABILITY)

    def get_name(self):
        return "Guarana"

    def create_child(self, x, y):
        return Guarana(self.world, x, y)

    def collision(self, attacker):
        self.die(attacker)
        attacker.increase_strength(3)  # Assuming `increase_strength` method exists in the Organism class
        print(f"{attacker.get_name()} now has strength: {attacker.get_strength()}")
        return True

File: PO_java/program/Organisms/Plants/Guarana.java
package projekt.program.Organisms.Plants;

import projekt.program.Organism;
import projekt.program.Organisms.Plant;
import projekt.program.World;

public class Guarana extends Plant {
    private static final int GUARANA_STRENGTH = 0; // Example strength for Guarana
    private static final int GUARANA_SEEDING_PROBABILITY = 10; // Example seeding probability

    public Guarana(World world, int x, int y) {
        super(world, x, y, GUARANA_STRENGTH, '^', GUARANA_SEEDING_PROBABILITY);
        // System.out.println("Creating Guarana");
    }

    @Override
    public String getName() {
        return "Guarana";
    }

    @Override
    public Organism createChild(int x, int y) {
        return new Guarana(this.world, x, y);
    }

    @Override
    public boolean collision(Organism attacker) {
        this.die(attacker);
        attacker.increaseStrength(3); // Assuming `increaseStrength` method exists in the Organism class
        System.out.println(attacker.getName() + " now has strength: " + attacker.getStrength());
        return true;
    }
}

